app/bundles/LeadBundle/Model/FieldModel.php

```diff
@@ -374,46 +374,54 @@ public function saveEntity($entity, $unlock = true)
             $entity->setIsListable(false);
         }
 
-        $event = $this->dispatchEvent('pre_save', $entity, $isNew);
-        $this->getRepository()->saveEntity($entity);
-        $this->dispatchEvent('post_save', $entity, $isNew, $event);
-
         $isUnique = $entity->getIsUniqueIdentifier();
 
-        if ($entity->getId()) {
-            //create the field as its own column in the leads table
-            $leadsSchema = $this->schemaHelperFactory->getSchemaHelper('column', $object);
-            if ($isNew || (!$isNew && !$leadsSchema->checkColumnExists($alias))) {
-                $schemaDefinition = self::getSchemaDefinition($alias, $entity->getType(), $isUnique);
-                $leadsSchema->addColumn(
+        //create the field as its own column in the leads table
+        $leadsSchema = $this->schemaHelperFactory->getSchemaHelper('column', $object);
+        if ($isNew || (!$isNew && !$leadsSchema->checkColumnExists($alias))) {
+            $schemaDefinition = self::getSchemaDefinition($alias, $type, $isUnique);
+            $leadsSchema->addColumn(
                     $schemaDefinition
                 );
+
+            try {
                 $leadsSchema->executeChanges();
+                $isCreated = true;
+            } catch (\Exception $e) {
+                $this->logger->addWarning($e->getMessage());
```

Comment:
As Don said before. You should filter the exceptions here and only attach the error message to the ones you are sure are relevant to this issue. I believe there are two of those: 
 - Error: 1118 SQLSTATE: 42000 (ER_TOO_BIG_ROWSIZE) - for exceeding max row size
 - Error: 1069 SQLSTATE: 42000 (ER_TOO_MANY_KEYS) - for exceeding max number of indexes

You do not want to catch all errors because that could give user the wrong idea of what the issue really is.